src/RequestHandlerSwooleRunner.php

```diff
@@ -81,43 +114,88 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
             };
 
         $this->swooleHttpServer = $swooleHttpServer;
+
+        $this->allowedStatic = $config['static_files'] ?? self::DEFAULT_STATIC_EXTS;
+        $this->docRoot = $config['options']['document_root'] ?? getcwd() . '/public';
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', function ($server) {
-            printf("Swoole is running at %s:%s\n", $server->host, $server->port);
-        });
-
-        $this->swooleHttpServer->on('request', function ($request, $response) {
-            printf(
-                "%s - %s - %s %s\n",
-                date('Y-m-d H:i:sO', time()),
-                $request->server['remote_addr'],
-                $request->server['request_method'],
-                $request->server['request_uri']
-            );
-            $emit = new SwooleEmitter($response);
-            try {
-                $psr7Request = ($this->serverRequestFactory)($request);
-            } catch (Throwable $e) {
-                // Error in generating the request
-                $this->emitMarshalServerRequestException($emit, $e);
-                return;
-            }
-            $emit->emit($this->handler->handle($psr7Request));
-        });
+        $this->swooleHttpServer->on('start', [$this, 'onStart']);
+        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
         $this->swooleHttpServer->start();
     }
 
+    /**
+     * On start event for swoole http server
+     */
+    public function onStart(SwooleHttpServer $server) : void
+    {
+        printf("Swoole is running at %s:%s\n", $server->host, $server->port);
```

Comment:
Psr logger would be a nice feature here.

Reply:
Yes, I think this can be a nice improvement for a future PR. I don't want to add more code here.